### ISSUE TITLE
Support detecting if a method is desctructor with estree

### ISSUE THREAD

**Issue by krysvac** [typescript-eslint/typescript-eslint#1032]
**What code were you trying to parse?**

```ts
ngOnDestroy(): void {
    *some code*
}
```

**What did you expect to happen?**
With the rule 
```json
"@typescript-eslint/explicit-member-accessibility": [
  "error",
  {
    "overrides": {
      "constructors": "off"
    }
  }
],
```
I was expecting it to not apply the rule to my destructor
**What actually happened?**
It applies the rule to the destructor forcing me to add an ignore statement on all destructors
**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.2` |
| `@typescript-eslint/parser` | `2.3.2` |
| `@typescript-eslint/typescript-estree` | `2.3.2` |
| `TypeScript`                           | `3.2.1` |
| `node`                                 | `8.10.0` |
| `npm`                                  | `6.11.2` |

**Comment by a-tarasyuk** [comment]
@krysvac what do you mean by `destructor`? `ngOnDestroy` is not a constructor. Maybe you need to use `methods` options instead?

**Comment by krysvac** [comment]
@a-tarasyuk You are correct in that it's not a constructor. I just found ignoredMethodNames which does what I wan't. I'll close this issue as it's invalid
